mpqp/execution/connection/aws_connection.py

```diff
@@ -49,48 +72,197 @@ def setup_aws_braket_account() -> tuple[str, list[Any]]:
         return "", []
 
 
+def update_aws_credentials_file(
+    profile_name: str,
+    access_key_id: str,
+    secret_access_key: str,
+    session_token: Optional[str],
+    region: str,
+):
+    """Create or update .aws/credentials file with the provided credentials.
+    It ensures the directory and file exist before making changes.
+    """
+
+    credentials_file = Path.home() / ".aws" / "credentials"
+
+    credentials_dir = credentials_file.parent
+    if not credentials_dir.exists():
+        credentials_dir.mkdir(parents=True, exist_ok=True)
+
+    config = ConfigParser()
+    if credentials_file.exists():
```

Comment:
if it doesn't exist, do we create the file ?

Reply:
Here, it effectively handles creating the ``aws/credentials`` file and the directory structure if they don't exist before adding the credentials for the specified profile. So, no modifications needed!!!
![image](https://github.com/user-attachments/assets/ea4aee51-a262-4bb7-8a98-bbb96736ad25)

Reply:
Why then do you need this condition line 94, and to read line 95 ?
To check if if has already the section with ``profile_name`` ?

Reply:
Because when the file exists, the code reads the existing contents into the ``ConfigParser`` object,  if not the ``ConfigParser `` won't attemept to read anything, it skips reading and the file will be created when writing the new configurations

Reply:
Then for me lines 97 and 98 should be also in side the ``if credentials_file.exists():``.

Reply:
Yeah correct, I think wont affect the code running but it’s unnecessary to check or remove sections from a non-existent file. No unnecessary checks keep the code clean and efficient.
will update it and check!!!

Reply:
Modified!